nand2tetris.el

```diff
@@ -59,32 +59,62 @@
+                                                                    ".sh"))
+                                                          (nand2tetris-tools-dir))))
 
 (defcustom nand2tetris-cpu-emulator nil
   "CPU Emulator Launcher."
   :group 'nand2tetris)
-(defun nand2tetris-cpu-emulator () (or nand2tetris-cpu-emulator (expand-file-name "CPUEmulator.sh" (nand2tetris-tools-dir))))
+(defun nand2tetris-cpu-emulator () (or nand2tetris-cpu-emulator
+                                       (expand-file-name (concat "CPUEmulator"
+                                                                 (if (eq system-type 'windows-nt)
+                                                                     ".bat"
+                                                                   ".sh"))
+                                                         (nand2tetris-tools-dir))))
 
 (defcustom nand2tetris-jack-compiler nil
   "Jack Compiler Launcher."
   :group 'nand2tetris)
-(defun nand2tetris-jack-compiler () (or nand2tetris-jack-compiler (expand-file-name "JackCompiler.sh" (nand2tetris-tools-dir))))
+(defun nand2tetris-jack-compiler () (or nand2tetris-jack-compiler
+                                        (expand-file-name (concat "JackCompiler"
+                                                                  (if (eq system-type 'windows-nt)
+                                                                      ".bat"
+                                                                    ".sh"))
+                                                          (nand2tetris-tools-dir))))
 
 (defcustom nand2tetris-text-comparer nil
   "Text Comparer Launcher."
   :group 'nand2tetris)
-(defun nand2tetris-text-comparer () (or nand2tetris-text-comparer (expand-file-name "TextComparer.sh" (nand2tetris-tools-dir))))
+(defun nand2tetris-text-comparer () (or nand2tetris-text-comparer
+                                        (expand-file-name (concat "TextComparer"
+                                                                  (if (eq system-type 'windows-nt)
+                                                                      ".bat"
+                                                                    ".sh"))
+                                                          (nand2tetris-tools-dir))))
 
 (defcustom nand2tetris-vm-emulator nil
   "VM Emulator Launcher."
   :group 'nand2tetris)
-(defun nand2tetris-vm-emulator () (or nand2tetris-vm-emulator (expand-file-name "VMEmulator.sh" (nand2tetris-tools-dir))))
+(defun nand2tetris-vm-emulator () (or nand2tetris-vm-emulator
+                                      (expand-file-name (concat "VMEmulator"
+                                                                (if (eq system-type 'windows-nt)
+                                                                    ".bat"
+                                                                  ".sh"))
+                                                        (nand2tetris-tools-dir))))
```

Comment:
Would you consider making a function for this following the naming conventions of this package?